test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -272,47 +268,57 @@ describe('<AddressSection>', () => {
       stateCode: '',
       type: ADDRESS_TYPES.domestic
     };
-    const tree = SkinDeep.shallowRender(<AddressSection {...props}/>);
+    const tree = shallow(<AddressSection {...props}/>);
 
-    const instance = tree.getMountedInstance();
-    expect(instance.state.isEditingAddress).to.be.true;
+    expect(tree.state('isEditingAddress')).to.be.true;
   });
 
   describe('validation', () => {
     // Spy on all the validation functions!
     beforeEach(spyOnValidators);
 
-    // Clean up the spies so nobody finds out
+    // Extract the spies so nobody finds out
     afterEach(cleanUpSpies);
 
     it('should run all validations against a field if the address is valid', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
+
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      // Change the city and blur for the validation to run
+      tree.find('input[name="city"]').simulate('change', { target: { value: 'Elsweyre' } });
+      tree.find('input[name="city"]').simulate('blur');
 
-      instance.handleChange('city', 'Elsweyre');
-      instance.dirtyInput('city');
       expect(AddressSection.fieldValidations.city.every(validator => validator.called)).to.be.true;
     });
 
     it('should return the first error message it finds', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
+
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      // Change the city and blur for the validation to run
+      tree.find('input[name="city"]').simulate('change', { target: { value: '' } });
+      tree.find('input[name="city"]').simulate('blur');
 
-      instance.handleChange('city', '');
-      instance.dirtyInput('city');
       // The required validator (first in the list) should return an error message and no other validators should run
       expect(AddressSection.fieldValidations.city[0].called).to.be.true;
       expect(AddressSection.fieldValidations.city.slice(1).every(validator => !validator.called)).to.be.true;
     });
 
     it('should run validations on modified fields only', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
 
-      const fieldsToModify = ['city', 'stateCode'];
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      const fieldsToModify = ['city', 'addressOne'];
```

Comment:
The naming on these fields probably needs overhauled; I changed this to `addressOne` instead of `stateCode` because the field's name is `state` but the associated variable is `stateCode`. This bit doesn't have anything to do with Enzyme.